app/src/ui/preferences/accessibility.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react'
+import { DialogContent } from '../dialog'
+import { Checkbox, CheckboxValue } from '../lib/checkbox'
+
+interface IAccessibilityPreferencesProps {
+  readonly underlineLinks: boolean
+  readonly onUnderlineLinksChanged: (value: boolean) => void
+}
+
+export class Accessibility extends React.Component<
+  IAccessibilityPreferencesProps,
+  {}
+> {
+  public constructor(props: IAccessibilityPreferencesProps) {
+    super(props)
+  }
+
+  public render() {
+    return (
+      <DialogContent>
+        <div className="advanced-section">
+          <h2>Accessibility</h2>
+          <Checkbox
+            label="Underline links"
+            value={
+              this.props.underlineLinks ? CheckboxValue.On : CheckboxValue.Off
+            }
+            onChange={this.onUnderlineLinksChanged}
+          />
+          <p className="git-settings-description">
+            When enabled, GitHub Desktop will underline links in commit
+            messages, comments, and other text fields. This can help make links
+            easier to distinguish. {this.renderExampleLink()}
+          </p>
+        </div>
+      </DialogContent>
+    )
+  }
+
+  private renderExampleLink() {
+    // The example link is rendered with inline style to override the global setting.
+    const style = {
+      textDecoration: this.props.underlineLinks ? 'underline' : 'none',
+    }
+
+    return (
+      // eslint-disable-next-line jsx-a11y/anchor-is-valid
+      <a className="link-button-component" style={style}>
```

Comment:
We should try avoid adding new linter disables if possible (we need to remove them for the scorecard).
The linter rule states:
```
The href attribute is required for an anchor to be keyboard accessible. Provide a valid, navigable address as the href value. If you cannot provide an href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/HEAD/docs/rules/anchor-is-valid.mdeslint[jsx-a11y/anchor-is-valid](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/anchor-is-valid.md)
```

I think we should switch this to either a `button` and add some styles to make it not appear as a button. Maybe worth asking the accessibility team if this could simply be a `span` since it is does not do anything -> it is strictly for visual purposes.